src/miner.cpp

```diff
@@ -417,16 +442,18 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
     int64_t nValueIn = 0;
     //Request all the coins here, check reserve later
 
-    if ( BalanceToStake<=0
-        || !wallet.SelectCoinsForStaking(BalanceToStake*2, txnew.nTime, CoinsToStake, nValueIn) )
```

Comment:
How does this actually solve the issue? In the old code, SelectCoinsForStaking would return false, causing ReasonNotStaking to be set and false returned from CreateCoinStake. The BreakForNoCoins does essentially the same. Or is the fix only in calling status.Clear() ?

Reply:
The fix for the linked issue is just to call `Clear()` to set the status variables back to zero because the UI checks those numbers for the staking indicator. I noticed that we say "No coins" for three different conditions, so I separated them to provide more informative messages (I imagine that someone might be confused if the wallet says "No coins" even though they do have coins, but they're just on cooldown). The `BreakForNoCoins` helper just does the same thing, but it wraps up the shared code so we can pass different messages without duplicating the logic.

Reply:
Thanks for clarifying that the Clear bit was the fix. I could not read it from the PR description (sorry). And I like how you separated the error cases!